ballerina/tests/data_binding_publish_tests.bal

```diff
@@ -617,6 +636,637 @@ service object {
+        if message is string {
+            receivedXmlPayloadValuePublish = message;
+            log:printInfo("Message Received: " + message);
+        }
+    }
+};
+
+@test:Config {
+    dependsOn: [testDataBindingXmlPayloadPublish],
+    groups: ["nats-basic"]
+}
+public function testDataBindingJsonPayloadPublish() {
+    json messageToSend = jsonData;
+    Client? newClient = clientObj;
+    if newClient is Client {
+        Listener? sub = listenerObj;
+        if sub is Listener {
+            checkpanic sub.attach(consumerServiceJsonPayloadPublish);
+            checkpanic sub.'start();
+            checkpanic newClient->publishMessage({ content: messageToSend, subject: DATA_BINDING_PUBLISH_SUBJECT });
+            int timeoutInSeconds = 120;
+            // Test fails in 2 minutes if it is failed to receive the message
+            while timeoutInSeconds > 0 {
+                if receivedJsonPayloadValuePublish !is () {
+                    json receivedMessage = receivedJsonPayloadValuePublish;
+                    test:assertEquals(receivedMessage, messageToSend, msg = "Message received does not match.");
+                    break;
+                } else {
+                    runtime:sleep(1);
+                    timeoutInSeconds = timeoutInSeconds - 1;
+                }
+            }
+            checkpanic sub.detach(consumerServiceJsonPayloadPublish);
+            if timeoutInSeconds == 0 {
+                test:assertFail("Failed to receive the message for 2 minutes.");
+            }
+        } else {
+            test:assertFail("NATS Connection creation failed.");
+        }
+    } else {
+        test:assertFail("NATS Connection creation failed.");
+    }
+}
+
+Service consumerServiceJsonPayloadPublish =
+@ServiceConfig {
+    subject: DATA_BINDING_PUBLISH_SUBJECT
+}
+service object {
+    remote function onMessage(json payload) {
+        receivedJsonPayloadValuePublish = payload;
+        log:printInfo("Message Received: " + payload.toJsonString());
+    }
+};
+
+@test:Config {
+    dependsOn: [testDataBindingJsonPayloadPublish],
+    groups: ["nats-basic"]
+}
+public function testDataBindingIntPayloadPublish() {
+    int messageToSend = 521;
+    Client? newClient = clientObj;
+    if newClient is Client {
+        Listener? sub = listenerObj;
+        if sub is Listener {
+            checkpanic sub.attach(consumerServiceIntPayloadPublish);
+            checkpanic sub.'start();
+            checkpanic newClient->publishMessage({ content: messageToSend, subject: DATA_BINDING_PUBLISH_SUBJECT });
+            int timeoutInSeconds = 120;
+            // Test fails in 2 minutes if it is failed to receive the message
+            while timeoutInSeconds > 0 {
+                if receivedIntPayloadValuePublish !is 0 {
+                    int receivedMessage = receivedIntPayloadValuePublish;
+                    test:assertEquals(receivedMessage, messageToSend, msg = "Message received does not match.");
```

Comment:
```suggestion
                    test:assertEquals(receivedIntPayloadValuePublish, messageToSend, msg = "Message received does not match.");
```

Reply:
Please check other places as well.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-nats/pull/427/commits/433577ffabea99c86a5ec34828b469bce521fc85